splinterd/src/config/builder.rs

```diff
@@ -23,15 +23,23 @@ pub trait PartialConfigBuilder {
     fn build(self) -> Result<PartialConfig, ConfigError>;
 }
 
-fn get_file_path(cert_dir: &str, file: &str) -> String {
+fn get_tls_file_path(config_dir: &str, cert_dir: &str, file: &str) -> String {
     if file.starts_with("./") || file.starts_with("../") {
         String::from(file)
-    } else {
+    } else if cert_dir.starts_with('/') || cert_dir.starts_with("./") || cert_dir.starts_with("../")
```

Comment:
A couple thoughts:

1. The absolute path check should be done with std::path::Path's absolute() check so it is cross-platform compatible.
2. The check for "./" and "../" should also check for the exact strings "." and ".." which would also be relative paths that we do not want to prefix. Really, then, we should split the path and (using stuff in std::path) and do this comparison on the first element of the path instead of start_with().